lib/foreman/telemetry_sinks/prometheus_sink.rb

```diff
@@ -1,41 +1,43 @@
-module Foreman::TelemetrySinks
-  class PrometheusSink
-    PROMETHEUS_STORE_DIR = File.join(Rails.root, 'tmp', 'prometheus')
+module Foreman
+  module TelemetrySinks
+    class PrometheusSink
+      PROMETHEUS_STORE_DIR = File.join(Rails.root, 'tmp', 'prometheus')
 
-    def initialize(opts = {})
-      require 'prometheus/client'
-      require 'prometheus/client/data_stores/direct_file_store'
-      # Set multiprocess-friendly data store
-      FileUtils.mkdir_p(PROMETHEUS_STORE_DIR)
-      # but clean it during startup as files will accumulate over time
-      FileUtils.rm_f(Dir.glob("#{PROMETHEUS_STORE_DIR}/*.bin"))
-      Prometheus::Client.config.data_store =
-        Prometheus::Client::DataStores::DirectFileStore.new(dir: PROMETHEUS_STORE_DIR)
-      @prom = ::Prometheus::Client.registry
-    end
+      def initialize(opts = {})
+        require 'prometheus/client'
+        require 'prometheus/client/data_stores/direct_file_store'
+        # Set multiprocess-friendly data store
+        FileUtils.mkdir_p(PROMETHEUS_STORE_DIR)
+        # but clean it during startup as files will accumulate over time
+        FileUtils.rm_f(Dir.glob("#{PROMETHEUS_STORE_DIR}/*.bin"))
+        Prometheus::Client.config.data_store =
+          Prometheus::Client::DataStores::DirectFileStore.new(dir: PROMETHEUS_STORE_DIR)
+        @prom = ::Prometheus::Client.registry
+      end
```

Comment:
Just to double check, this code must be executed BEFORE processes are forked from the parent process. Otherwise this would delete all bin files on each process restart of the cluster. Does it make sense? I think I need to test this.

Reply:
It should be executed the same time as it did before, tho the order of the Telemetry calls slighly changed, but I didn't think it would be anything significant.